modules/gitops-release-notes-1-17-0.adoc

```diff
@@ -0,0 +1,149 @@
+====
+
+[id="gitops-1-17-0-fixed-UI-error-for-http-based-repository-credentials_{context}"]
+=== Fixed UI error when editing HTTP-based repository credentials
+
+* Before this update, the Argo CD Web UI failed to handle repositories hosted over HTTP due to URL validation errors. This prevented users from editing credentials for HTTP-based repositories through the Web UI. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]
+
+[id="gitops-1-17-0-authentication-tokens-redacted-in-argo-CD-e2e-test-logs_{context}"]
+=== Authentication tokens redacted in Argo CD E2E test logs
+ 
+* Before this update, Argo CD end-to-end (E2E) tests exposed authentication tokens in logs due to missing command argument redaction. This posed a security risk by potentially compromising authentication tokens during test execution. With this update, an authentication token redactor has been integrated into E2E test command execution. Authentication tokens are now hidden in logs, enhancing test security and protecting user credentials. link:https://issues.redhat.com/browse/GITOPS-3942[GITOPS-3942]
+
+[id="deprecated-features-1-17-0_{context}"]
+== Deprecated and removed features
+
+[id="deprecation-of-some-fields-in-argo-cd_{context}"]
+=== Deprecation of Keycloak support in OpenShift GitOps
+
+* In {gitops-title} 1.17, the support for Keycloak-based authentication is deprecated. When Keycloak is configured for single sign-on (SSO), deprecation warnings appear in the logs. Dex is the default and recommended authentication provider. It integrates with the OpenShift OAuth server and supports authentication using OpenShift users and groups. Users are required to migrate to Dex-based authentication for continued support. To continue using Keycloak, you must deploy and manage their own link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/operator_guide/installation-[Red Hat Build of Keycloak (RHBK)] instance and manually configure the integration in the Argo CD custom resource. link:https://issues.redhat.com/browse/GITOPS-7070[GITOPS-7070]
+
+[id="removed-support-for-deprecated-initialRepositories-repositoryCredentials-fields_{context}"]
+=== Removed support for deprecated `initialRepositories` and `repositoryCredentials` fields
+
+* In {gitops-title} 1.17, support for the deprecated `initialRepositories` and `repositoryCredentials` fields has been removed. If these fields are used, Argo CD logs a warning indicating their removal. Users can add repositories and repository credentials manually using the Argo CD CLI or the Argo CD dashboard. link:https://issues.redhat.com/browse/GITOPS-5961[GITOPS-5961]
+
+[id="breaking-change-1-17-0_{context}"]
+== Breaking change
+
+[id="rbac-behavior-changes-with-argo-cd-dex-sso_{context}"]
+=== RBAC behavior changes with Argo CD 3.0 and Dex SSO
```

Comment:
Can we add one section that says `Major version upgrade of Argo CD 3.0` and then add these breaking changes as sub sections. Also we can point to this upstream [link](https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.14-3.0/) for more details on the breaking changes and its mitigation. WDYT ?

Reply:
+1